### PR TITLE
Implement new plot directive for clustering visualizers [Issue #687]

### PR DESCRIPTION
**Pull request by Kautumn06**
Adds the new plot directive from PR #446 in the documentation for the `KElbowVisualizer`, `SilhouetteVisualizer`, and `InterclusterDistance`. In addition, I deleted the scripts originally used to generate the images in the docs, along with the images.

I'll add comments to help explain the changes. However, please just let me know if anyone has any questions!

Quick Update: the tests originally failed which I think may have been because I initially set `k` in the `InterclusterDistance` visualizer, similar to how `k` is set in `KElbowVisualizer`. 
```
model = KMeans()
visualizer = InterclusterDistance(model, k=9)
```

And while it did create the image when I ran `make html`, the tests failed. I think this may be because `k` is not actually a defined parameter for the `InterclusterDistance` visualizer, which is why I pushed a new commit with the following update:
```
model = KMeans(9)
visualizer = InterclusterDistance(model)
```

### CODE REVIEW

File: docs/api/cluster/elbow.rst [hunk 1]
```diff
@@ -8,21 +8,23 @@ The ``KElbowVisualizer`` implements the "elbow" method to help data scientists s
 To demonstrate, in the following example the ``KElbowVisualizer`` fits the ``KMeans`` model for a range of :math:`K` values from 4 to 11 on a sample two-dimensional dataset with 8 random clusters of points. When the model is fit with 8 clusters, we can see an "elbow" in the graph, which in this case we know to be the optimal number.
 
 .. plot::
+    :context: close-figs
```
Comment (Kautumn06) [inline docs/api/cluster/elbow.rst]: Added the recommended `:context:` and `alt` settings to the plot directive.

File: docs/api/cluster/elbow.rst [hunk 2]
```diff
@@ -8,21 +8,23 @@ The ``KElbowVisualizer`` implements the "elbow" method to help data scientists s
...
 
     from sklearn.cluster import KMeans
     from sklearn.datasets import make_blobs
 
     from yellowbrick.cluster import KElbowVisualizer
 
-    # Create synthetic dataset with 8 random clusters
-    X, y = make_blobs(centers=8, n_features=12, shuffle=True, random_state=42)
+    # Generate synthetic dataset with 8 random clusters
+    X, y = make_blobs(n_samples=1000, n_features=12, centers=8, random_state=42)
```
Comment (Kautumn06) [inline docs/api/cluster/elbow.rst]: I removed `shuffle=True` since it is `True` by default.

In addition, I added `n_samples=1000` since that is what it was set to in the original script used to generate the image, and if not included, `n_samples` is only equal to `100` by default.

File: docs/api/cluster/elbow.rst [hunk 3]
```diff
@@ -8,21 +8,23 @@ The ``KElbowVisualizer`` implements the "elbow" method to help data scientists s
...
 
     # Instantiate the clustering model and visualizer
     model = KMeans()
     visualizer = KElbowVisualizer(model, k=(4,12))
 
-    visualizer.fit(X)    # Fit the data to the visualizer
-    visualizer.poof()    # Draw/show/poof the data
+    visualizer.fit(X)        # Fit the data to the visualizer
```
Comment (Kautumn06) [inline docs/api/cluster/elbow.rst]: I like to add 8 spaces between the code and inline comments in the docs, since I think it makes it easier to read.

File: docs/api/cluster/elbow.rst [hunk 4]
```diff
@@ -31,21 +33,25 @@ However, two other metrics can also be used with the ``KElbowVisualizer`` -- ``s
 The ``KElbowVisualizer`` also displays the amount of time to train the clustering model per :math:`K` as a dashed green line, but is can be hidden by setting ``timings=False``. In the following example, we'll use the ``calinski_harabaz`` score and hide the time to fit the model.
 
 .. plot::
+    :context: close-figs
```
Comment (Kautumn06) [inline docs/api/cluster/elbow.rst]: Added new plot directive settings.

File: docs/api/cluster/elbow.rst [hunk 5]
```diff
@@ -31,21 +33,25 @@ However, two other metrics can also be used with the ``KElbowVisualizer`` -- ``s
...
 
     from sklearn.cluster import KMeans
     from sklearn.datasets import make_blobs
 
     from yellowbrick.cluster import KElbowVisualizer
 
-    # Create synthetic dataset with 8 random clusters
-    X, _ = make_blobs(centers=8, n_features=12, shuffle=True, random_state=42)
+    # Generate synthetic dataset with 8 random clusters
+    X, y = make_blobs(n_samples=1000, n_features=12, centers=8, random_state=42)
```
Comment (Kautumn06) [inline docs/api/cluster/elbow.rst]: I replaced `_` with `y` here, since even though the target variable is not needed and it doesn't affect the final plot, I thought `_` could be confusing for new users who may not have seen this before. In addition, it's not used in the other examples in our documentation when `make_blobs` is used to create the dataset.

I also removed `shuffle=True` and added `n_samples=1000` for the same reasons I described above in a previous comment.

File: docs/api/cluster/elbow.rst [hunk 6]
```diff
@@ -31,21 +33,25 @@ However, two other metrics can also be used with the ``KElbowVisualizer`` -- ``s
...
     # Instantiate the clustering model and visualizer
     model = KMeans()
-    visualizer = KElbowVisualizer(model, k=(4,12), metric='calinski_harabaz', timings=False)
+    visualizer = KElbowVisualizer(
+        model, k=(4,12), metric='calinski_harabaz', timings=False
+    )
 
-    visualizer.fit(X)    # Fit the data to the visualizer
-    visualizer.poof()    # Draw/show/poof the data
+    visualizer.fit(X)        # Fit the data to the visualizer
```
Comment (Kautumn06) [inline docs/api/cluster/elbow.rst]: Added additional spacing.

File: docs/api/cluster/elbow.rst [hunk 7]
```diff
@@ -31,21 +33,25 @@ However, two other metrics can also be used with the ``KElbowVisualizer`` -- ``s
...
 
-    # Create synthetic dataset with 8 random clusters
-    X, _ = make_blobs(centers=8, n_features=12, shuffle=True, random_state=42)
+    # Generate synthetic dataset with 8 random clusters
+    X, y = make_blobs(n_samples=1000, n_features=12, centers=8, random_state=42)
 
     # Instantiate the clustering model and visualizer
     model = KMeans()
-    visualizer = KElbowVisualizer(model, k=(4,12), metric='calinski_harabaz', timings=False)
+    visualizer = KElbowVisualizer(
```
Comment (Kautumn06) [inline docs/api/cluster/elbow.rst]: Small formatting change since the line was quite long.

File: docs/api/cluster/icdm.rst [hunk 8]
```diff
@@ -5,26 +5,24 @@ Intercluster Distance Maps
 
 Intercluster distance maps display an embedding of the cluster centers in 2 dimensions with the distance to other centers preserved. E.g. the closer to centers are in the visualization, the closer they are in the original feature space. The clusters are sized according to a scoring metric. By default, they are sized by membership, e.g. the number of instances that belong to each center. This gives a sense of the relative importance of clusters. Note however, that because two clusters overlap in the 2D space, it does not imply that they overlap in the original feature space.
 
-.. code:: python
+.. plot::
```
Comment (Kautumn06) [inline docs/api/cluster/icdm.rst]: Previously, this example in the documentation had been broken out into two blocks&mdash;one to generate the dataset and one to `fit` the model and `poof` the visualizer. So in order to have it work with the new plot directive, I combined them into one and moved all of the import statements to the top of the block.

File: docs/api/cluster/icdm.rst [hunk 9]
```diff
@@ -5,26 +5,24 @@ Intercluster Distance Maps
...
-    # Make 12 blobs dataset
-    X, y = make_blobs(centers=12, n_samples=1000, n_features=16, shuffle=True)
-
-.. code:: python
-
-    from sklearn.cluster import KMeans
     from yellowbrick.cluster import InterclusterDistance
 
-    # Instantiate the clustering model and visualizer
-    visualizer = InterclusterDistance(KMeans(9))
```
Comment (Kautumn06) [inline docs/api/cluster/icdm.rst]: As you'll see in the next comment, I broke this code out into two lines to keep it consistent with the rest of the examples in the documentation.

File: docs/api/cluster/icdm.rst [hunk 10]
```diff
@@ -5,26 +5,24 @@ Intercluster Distance Maps
...
-
-    visualizer.fit(X) # Fit the training data to the visualizer
-    visualizer.poof() # Draw/show/poof the data
+    # Generate synthetic dataset with 12 random clusters
+    X, y = make_blobs(n_samples=1000, n_features=12, centers=12, random_state=42)
 
+    # Instantiate the clustering model and visualizer
```
Comment (Kautumn06) [inline docs/api/cluster/icdm.rst]: This code had previously been combined in a single line, and while that does work, I think it's clearer to have it in two. In addition, this is the format we could throughout the rest of the documentation so I wanted to be consistent.

File: docs/api/cluster/icdm.rst [hunk 11]
```diff
@@ -5,26 +5,24 @@ Intercluster Distance Maps
...
-    from sklearn.cluster import KMeans
     from yellowbrick.cluster import InterclusterDistance
 
-    # Instantiate the clustering model and visualizer
-    visualizer = InterclusterDistance(KMeans(9))
-
-    visualizer.fit(X) # Fit the training data to the visualizer
-    visualizer.poof() # Draw/show/poof the data
+    # Generate synthetic dataset with 12 random clusters
+    X, y = make_blobs(n_samples=1000, n_features=12, centers=12, random_state=42)
```
Comment (Kautumn06) [inline docs/api/cluster/icdm.rst]: Originally, the `.rst` file showed `n_features=16`; however, in the script used to generate the image, the parameter was set to `n_features=12`, so I changed it here to reflect that.

File: docs/api/cluster/silhouette.rst [hunk 12]
```diff
@@ -7,29 +7,23 @@ The Silhouette Coefficient is used when the ground-truth about the dataset is un
...
-    X, y = make_blobs(centers=8)
-
-.. code:: python
-
-    from sklearn.cluster import MiniBatchKMeans
-
     from yellowbrick.cluster import SilhouetteVisualizer
 
+    # Generate synthetic dataset with 8 random clusters
+    X, y = make_blobs(n_samples=1000, n_features=12, centers=8, random_state=42)
```
Comment (Kautumn06) [inline docs/api/cluster/silhouette.rst]: Added the parameters used in the original script to generate the image.

File: docs/api/cluster/silhouette.rst [hunk 13]
```diff
@@ -7,29 +7,23 @@ The Silhouette Coefficient is used when the ground-truth about the dataset is un
...
+
     # Instantiate the clustering model and visualizer
-    model = MiniBatchKMeans(6)
+    model = KMeans(6)
```
Comment (Kautumn06) [inline docs/api/cluster/silhouette.rst]: Using `KMeans` instead of `MiniBatchKMeans` didn't affect the plot so I changed it here to make it consistent with our other clustering visualizers. In addition, I was worried that new users might be confused about why it is used here and not in the other clustering examples.

**Comment by rebeccabilbro** [comment]
Hi @Kautumn06 &mdash; sorry that I didn't get to this earlier today, but we've been sorting out some problems with our automated tests. Now that everything seems to be fixed, I've updated your branch so that the tests can run again; I'll take a look at your updates tomorrow!

**Comment by Kautumn06** [comment]
Hi @rebeccabilbro no problem! I haven't pushed any changes since yesterday afternoon so hopefully there shouldn't be anymore problems. However, just let me know if you have any questions or if there is anything else you need me to help with!

**Comment by rebeccabilbro** [comment]
Hi @Kautumn06 - I'm starting to take a look at your updates to the clustering docs and I had a quick question &mdash; in your above message, you said:

> when I ran `make html`, the tests failed.

and I just wanted to make sure I understood what behavior you were experiencing before proceeding. 

When you run `make html`, it builds a copy of the docs locally on your machine, but it doesn't run any tests. However, oftentimes when we run the `make html` command after making experimental changes to the docs, we'll get warnings or error messages from Sphinyx in the command line that tell us that there's something funky going on in the docs. When you say the tests failed after you ran `make html`, do you mean you got Sphinyx warnings? This is probably a sign that there is some kind of formatting error in the rst, a missing reference link, etc.

Alternatively, if you mean that after making the updates to the docs and running the tests (either locally on your machine with `pytest`, or by opening the PR, which automatically runs the tests and shows the results here on Github), you observed failing tests on the command line or saw the little red X mark here on GH, that's possibly due to the [miniconda/Appveyor problems we were experiencing](https://github.com/DistrictDataLabs/yellowbrick/pull/743) with our continuous integration tools). 

The continuous integration testing issue is now resolved (for now ;D ), so would you please let me know if what you experienced seemed instead like a Sphinx issue, so I can look into that during my code review? Thanks for working on this!

**Comment by Kautumn06** [comment]
Hi @rebeccabilbro sorry for any confusion! The test I was referring to was in fact the red X here on GitHub. Originally, I had thought that the problem occurred because of the parameter issue I mentioned above; however, it actually turned out to be from the [miniconda/Appveyor problem](https://github.com/DistrictDataLabs/yellowbrick/pull/743) we've been experiencing.

So please just let me know if that answers your question or if there is anything else you need me to do!

**Comment by Kautumn06** [comment]
Hi @rebeccabilbro  I just wanted to let you know that I fixed the two minor merge conflicts that came up today after #729 was merged in. It was a quick fix since the two differences were from me including the new plot directives `:alt:` option in the code blocks. Let me know if you have any questions!

**Comment by Kautumn06** [comment]
Hi @rebeccabilbro I've replaced `KMeans` with `MiniBatchKMeans` in the `SilhouetteVisualizer` documentation example. I also added the plot directive `:alt:` option to the code block since I had originally forgot to add it. Once the tests have passed, I'll merge it in.
